src/Components/SysTable/hooks.js

```diff
@@ -0,0 +1,135 @@
+import { useApolloClient } from '@apollo/client';
+import { useEffect, useLayoutEffect } from 'react';
+import { useDispatch } from 'react-redux';
+
+export const useInventoryUtilities = (
+    inventory
+) => {
+    const dispatch = useDispatch();
+
+    // Resets the Inventory to a loading state
+    // and prevents previously shown columns and rows to appear
+    useLayoutEffect(() => {
+        dispatch({
+            type: 'INVENTORY_INIT'
+        });
+    }, []);
+
+    // Filters do not yet trigger the inventory to call getEntities
+    // and the page would not reset to page 1
+    const resetPage = () => {
+        Promise.resolve(
+            dispatch({
+                type: 'RESET_PAGE'
+            })
+        ).then(() => inventory?.current?.onRefreshData());
```

Comment:
I don't think we have to wrap the `dispatch` with `Promise`
```suggestion
        dispatch({
            type: 'RESET_PAGE'
        })
        inventory?.current?.onRefreshData()
```